streaming/src/main/scala/org/apache/spark/streaming/dstream/FileInputDStream.scala

```diff
@@ -74,10 +74,12 @@ class FileInputDStream[K, V, F <: NewInputFormat[K,V]](
     directory: String,
     filter: Path => Boolean = FileInputDStream.defaultFilter,
     newFilesOnly: Boolean = true,
-    conf: Option[Configuration] = None)
+    @transient conf: Option[Configuration] = None)
```

Comment:
I've always disliked using `@transient` for this kind of thing ... can it just be a constructor param, that is only used to construct `serializableConfOpt`?